### ISSUE TITLE
Dealing with student runtime errors in __name__ == '__main__' blocks

### ISSUE THREAD

**Issue by acbart** [pedal-edu/pedal#67]
Pedal runs student code, in order to determine if there were runtime errors. When it does so, everything is exactly the same as when you run student code normally, with only small differences. One such difference: the `__name__` variable gets the value `answer.py` instead of `__main__`. Because that's the purpose of the `__name__` variable, to let you determine that the code is being run by someone else. 

Unfortunately, if the student has an error in that block (perhaps calling some of their other functions), then Pedal never sees the error. This leads to the unfortunate scenario where there's no high-priority runtime error, so other things fail in weird ways (e.g., code coverage check reports that you covered 0% of your code).

A hacky solution is apparently as follows, at least in BlockPy:

```python
student = get_sandbox()
student.data['__name__'] = '__main__'
student = run()
```

In other words, we rerun their code with the `__name__` manually set. We probably need a more robust solution in the Sandbox for this.
